### PR TITLE
Added the option to config the server endpoint

### PR DESCRIPTION
**Pull request by welsinga**

### CODE REVIEW

File: library/src/main/java/nl/surfnet/nonweb/sso/SSOCallback.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * An {@code SSOCallback} represents the callback handler to be invoked
+ * when an asynchronous call is completed or it fails.
+ *
+ * @author W.Elsinga
+ */
+public interface SSOCallback {
+    /**
+     * @param credential {@link Credential}
```
Comment (dzolnai) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOCallback.java:32]: Some missing comments on the methods here.

File: library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java [hunk 2]
```diff
@@ -62,29 +48,50 @@ public void onCreate(Bundle savedInstanceState) {
         if (BuildConfig.DEBUG) {
             Log.d(TAG, "Starting task to retrieve token.");
         }
-        new RequestTokenTask(this, _consumerId).execute();
+        new RequestTokenTask(this, //
```
Comment (dzolnai) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java]: No need to put the parameters in multiple lines if it wouldn't exceed line length

File: library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java [hunk 3]
```diff
@@ -62,29 +48,50 @@ public void onCreate(Bundle savedInstanceState) {
...
 
 
     /**
      * Authorize the {@code consumerId}
      *
+     * @param context    the context
+     * @param consumerId consumer ID to be validated
+     * @param endpoint   oauth-server endpoint
+     */
+    public static void authorize(@NonNull final Context context, //
```
Comment (dzolnai) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java:62]: Why the empty comments on the end of the lines?

File: library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java (hunk 2 above)
Comment (welsinga) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java]: True, but better readable

Comment (dzolnai) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java]: I find it harder to read this way, I guess it is subjective :)
You don't have to fix this, was just a note

Comment (welsinga) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java]: But better readable

File: library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java (hunk 3 above)
Comment (welsinga) [inline library/src/main/java/nl/surfnet/nonweb/sso/SSOServiceActivity.java:62]: Just a convenience method.

**Pull request merged**